app/workers/facilities/mental_health_reload_job.rb

```diff
@@ -19,13 +19,9 @@ def fetch_mental_health_data
       raise MentalHealthDownloadError, "Failed to download mental health data: #{e.cause}"
     end
 
-    def update_cache(model, records)
+    def update_cache(records)
       records.each do |r|
-        ext = if r['Extension'] == '0' || r['Extension'] == 'NULL'
-                nil
-              else
-                r['Extension']
-              end
+        ext = valid_extension?(r['Extension']) ? r['Extension'] : nil
```

Comment:
Ok, now I think this is harder to understand.

I thought you were excluding all the records where the extension was '0' or 'NULL' but you're doing a *default* of `nil`. Its a lot harder to read like this for sure. How about something like

```
        attrs = {
          station_number: r['StationNumber'],
          mh_phone: r['MHPhone'],
          mh_ext: ensure_valid_extension(r['Extension']),
          modified: r['Modified'],
          local_updated: Time.now.utc.iso8601
        }

....

def ensure_valid_extension(extension:)
  if extension == '0' || extension == 'NULL'
                nil
      else
               extension
      end
end
```

Reply:
Interesting. Thinking deeper about it, I think the code's true function is converting from one encoding to another. That is, encoding how to denote "no extension extension":

* In the source datastore, "none" is represented by the strings `"0"` or `"NULL"`.
* In the destination datastore, "none" is represented by the object `nil`.

And so I'd vote for making this explicit:

```ruby
EXTENSION_MAPPINGS = {
  '0' => nil,
  'NULL => nil
}

# ...

def convert_extension(ext)
  return ext unless EXTENSION_MAPPINGS.key?(ext)

  EXTENSION_MAPPINGS[ext]
end
```

Reply:
Similar idea and does the same job of moving the logic out of the way and into a method (which was the original issue w/ the CodeClimate warning).

I think the `if/else` is way more braindead (a plus) and thus easier to understand when someone reads this a year from now but the map is pretty readable too and  makes a lot of sense, esp for a more complex mapping for sure @dogweather

Reply:
I'm gonna go with the if statement. I like the idea of confining all the logic to one method